objects/Door.js

```diff
@@ -0,0 +1,74 @@
+const DOOR_TRIGGERS = {
+    AUTOMATIC: 'auto',
+    UP: 'up',
+    DOWN: 'down'
+}
+class Door extends FinishFlag {
+    constructor(x, y, tileSize, type, tilemapHandler, extraAttributes = {}) {
+        super(x, y, tileSize, type, tilemapHandler, extraAttributes);
+        
+        // overrides
+        this.lockedSpriteIndex = SpritePixelArrays.getIndexOfSprite(ObjectTypes.LOCKED_DOOR);
+        this.lockedSpriteYPos = this.lockedSpriteIndex * this.tileSize;
+        this.sound = SoundHandler.door;
+
+        // new
+        if(!extraAttributes?.activationTrigger) {
+            this.addChangeableAttribute("activationTrigger", DOOR_TRIGGERS.UP);
+        }
+        if(!extraAttributes?.flagIndex) {
+            this.flagIndex = this.makeid(3);
+            this.addChangeableAttribute("flagIndex", this.flagIndex, this.tilemapHandler.currentLevel);
+        }
+    }
+
+    collisionEvent() { // overrides super.collisionEvent
+        if (this.locked) {
+            return;
+        }
+        PlayerInteractionHandler.registerInteractionTarget(this);
+    }
+
+    changeExit(customExit, reciprocatingLevelIndex) {
+        super.changeExit(customExit, reciprocatingLevelIndex);
+        
+        // Door-specific effect: if exit target is a door, connect it back to this
+        // (make the connection mutual)
+        let targetLevelIndex = customExit.levelIndex;
+        let targetObjectType = customExit.type;
+        let targetFlagIndex = customExit.flagIndex;
```

Comment:
try not to reassign variables where there is no need for please. just use customExit.type where you need it down below f.e.

Reply:
Sure, if you prefer. These local variables are just for readability. I feel like the naming here otherwise requires a broader awareness/understanding of other code and would be a bit harder to jump into.